### PR TITLE
SDK 2.0: AssetTree

### LINKED ISSUE #313: Move AssetTree component to new SDK
Create a new branch from https://github.com/cognitedata/gearbox.js/tree/sdk-2.0
After finishing with the task – create PR to this brunch and wait for approving.

### PR DESCRIPTION
**Pull request by joyrostie**
* Updated `AssetTree` to SDK 2.0
* Updated `AssetTree` story and unit tests
* Updated `ThemeProvider` story for `AssetTree` 
* Updated mocks `assetListV2`

Closes #313

### CODE REVIEW

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-342.gearboxjs.preview.cogniteapp.com

File: src/components/AssetTree/AssetTree.tsx [hunk 1]
```diff
@@ -105,16 +95,28 @@ export class AssetTree extends Component<AssetTreeProps, AssetTreeState> {
...
-          : [],
+        assets && assets.length > 0 ? AssetTree.mapDataAssets(assets) : [],
     });
   }
 
+  cursorApiRequest = async (assetId: number): Promise<Asset[]> => {
+    const result = await this.context!.assets.list({
+      filter: { parentIds: [assetId] },
+    }).autoPagingToArray();
+    if (!result && !Array.isArray(result)) {
```
Comment (xitman78) [inline src/components/AssetTree/AssetTree.tsx]:
```suggestion
    if (!result || !Array.isArray(result)) {
```

File: src/components/AssetTree/AssetTree.tsx [hunk 2]
```diff
@@ -105,16 +95,28 @@ export class AssetTree extends Component<AssetTreeProps, AssetTreeState> {
...
+      console.error(ERROR_API_UNEXPECTED_RESULTS);
+    }
+    return [...result];
```
Comment (xitman78) [inline src/components/AssetTree/AssetTree.tsx]: Why do we clone the array here?

File: src/components/AssetTree/AssetTree.tsx (hunk 1 above)
Comment (joyrostie) [reply src/components/AssetTree/AssetTree.tsx]: Fixed

File: src/components/AssetTree/AssetTree.tsx (hunk 2 above)
Comment (joyrostie) [reply src/components/AssetTree/AssetTree.tsx]: Nope, it's not necessary, changed to return fetched array.

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-342.gearboxjs.preview.cogniteapp.com

**Pull request merged**